### PR TITLE
Disconnect origin when send to client was failed

### PR DESCRIPTION
**Pull request by heat1024**
This PR makes 2 changes as follows...

## Make disconnect origin connection when send to client was failed

[In previous PR](https://github.com/pyama86/pftp/pull/30), I was delete `proxy.Close()` function from defer in `client_handler.go` file.

But, It makes error when client blocked response packet from server.
(In that case, goroutine in proxy.go will never end).

So this PR makes disconnect origin proxy connection when send response to client was failed.
(I think it was necessary, but missing before 😓 )

## Set linger to each tcp connections.

pftp makes 3 connection for one client connection. (client, local ftp and origin ftp)

It makes a lot of of TIME_WAIT to pftp server. It means pftp use a lot of socket resources.
Resolve this problem, add SetLinger to 0 when create connection.

### CODE REVIEW

**Comment by heat1024** [comment]
Added keepalive_time to config file(in last commit, it had hard coded by 15 sec for testing 😓 )

At default, the KeepaliveTime variable will set by 900 sec.

```Diff
...
transfer_timeout = 600
+keepalive_time = 600
...
```

```Diff
// set linger 0 and tcp keepalive setting between client connection
		conn.SetKeepAlive(true)
-		conn.SetKeepAlivePeriod(time.Duration(15) * time.Second)
+		conn.SetKeepAlivePeriod(time.Duration(server.config.KeepaliveTime) * time.Second)
		conn.SetLinger(0)
```

File: pftp/client_handler.go [hunk 1]
```diff
@@ -84,7 +84,6 @@ func (c *clientHandler) setClientDeadLine(t int) {
 }
 
 func (c *clientHandler) handleCommands() error {
-	proxyError := make(chan error)
```
Comment (pyama86) [inline pftp/client_handler.go]: I can't understand why do you delete `proxyError`?

File: pftp/proxy.go [hunk 2]
```diff
@@ -244,7 +250,7 @@ func (s *proxyServer) switchOrigin(clientAddr string, originAddr string, tlsProt
 			s.log.debug("reconnect to origin")
 			cnt++
 
-			s.origin.Close()
```
Comment (pyama86) [inline pftp/proxy.go]: Could you tell me the meaning of the change?

Comment (heat1024) [reply pftp/proxy.go]: It's just same with s.origin.Close() but want to match the function used in client_handler.

File: pftp/client_handler.go (hunk 1 above)
Comment (heat1024) [reply pftp/client_handler.go]: Oh, it just missed.

I tested a few things what channels makes hang sometimes.
But after end of test, I forgot this channel was deleted.

Thanks for find that!

Comment (heat1024) [reply pftp/client_handler.go]: reverted : https://github.com/pyama86/pftp/pull/31/commits/a4f90b5b2aa4e270f8963be563c9fc7f99d4a5e2

**Comment by pyama86** [comment]
thanks!

**Comment by heat1024** [comment]
Thanks for merge 😄

**Pull request merged**